libvast/src/schema.cpp

```diff
@@ -217,12 +217,16 @@ caf::expected<schema> load_schema(const std::vector<path>& schema_paths) {
           case path::regular_file:
           case path::symlink: {
             auto schema = load_schema(f);
-            if (!schema)
-              return schema.error();
+            if (!schema) {
+              VAST_ERROR_ANON(__func__, render(schema.error()), f);
+              continue;
+            }
             if (auto merged = schema::merge(directory_schema, *schema))
               directory_schema = std::move(*merged);
-            else
-              return make_error(ec::format_error, "type clash in schema");
+            else {
+              VAST_ERROR_ANON(__func__, "type clash in schema");
```

Comment:
In this case, we should also inform the user about the name of the type with a duplicate definition as well as the paths to both schema files.

I'm also not sure if it is really sensible to continue in this case.

Reply:
I've done the necessary changes.

The output is now this for type clashes:

```
2020-06-19T12:29:52.205 failed to read schema dirs: !! format_error: ("type clash: cannot accomodate two types with the same name:", "suricata.component.common") "in schema file" "schema/suricata.schema"
```

Reply:
Note that I cannot know which schema the type we clashed with was from, as that is already in the merged schema.

Reply:
Let's make a note to consider switching to a 2-phase ingestion process for this.

Reply:
I'll create a new story.